src/protocols/fiat-protocol.js

```diff
@@ -0,0 +1,217 @@
+ */
+
+/**
+ * A protocol-agnostic, standardized object representing a supported crypto asset.
+ * @typedef {Object} FiatSupportedAsset
+ * @property {string} code -Provider-specific asset code for the crypto asset.
+ * @property {string} networkCode - The network code for the asset, if applicable (e.g., 'ethereum', 'tron').
+ * @property {number} precision - The number of decimal places for the asset.
+ * @property {string} [name] - The asset's full name (e.g., 'Bitcoin').
+ * @property {Record<string, any>} [metadata] - Provider-specific raw data for this asset.
+ */
+
+/**
+ * A protocol-agnostic, standardized object representing a supported fiat currency.
+ * @typedef {Object} FiatSupportedCurrency
+ * @property {string} code - The currency's ISO 4217 code (e.g., 'USD').
+ * @property {number} precision - The number of decimal places for the currency.
+ * @property {string} [name] - The currency's full name (e.g., 'United States Dollar').
+ * @property {Record<string, any>} [metadata] - Provider-specific raw data for this currency.
+ */
+
+/**
+ * A protocol-agnostic, standardized object representing a supported country.
+ * @typedef {Object} FiatSupportedCountry
+ * @property {string} code - The country's ISO 3166-1 alpha-2 or alpha-3 code.
+ * @property {boolean} isBuyAllowed - Whether buying is supported in this country.
+ * @property {boolean} isSellAllowed - Whether selling is supported in this country.
+ * @property {string} [name] - The country's common name.
+ * @property {Record<string, any>} [metadata] - Provider-specific raw data for this region.
+ */
+
+/**
+ * @interface
+ */
+export class IFiatProtocol {
+  /**
+   * Generates a widget URL for a user to purchase a crypto asset with fiat currency.
+   * @param {string} cryptoAsset - The provider-specific code of the crypto asset to purchase.
+   * @param {string} fiatCurrency - The currency's ISO 4217 code (e.g., 'USD').
+   * @param {number} amount - The amount of crypto asset to buy, in its main unit (e.g., 1.50 for 1.50 ETH).
+   * @param {string} [recipient] - The wallet address to receive the purchased crypto asset.
+   * @param {Record<string, any>} [config] - Provider-specific configuration for the buy operation.
+   * @returns {Promise<string>} The URL for the user to complete the purchase.
+   */
+  async buy (cryptoAsset, fiatCurrency, amount, recipient, config) {
+    throw new NotImplementedError('buy(cryptoAsset, fiatCurrency, amount, recipient, config)')
+  }
+
+  /**
+   * Generates a widget URL for a user to sell a crypto asset for fiat currency.
+   * @param {string} cryptoAsset - The provider-specific code of the crypto asset to sell.
+   * @param {string} fiatCurrency - The currency's ISO 4217 code (e.g., 'USD').
+   * @param {number} amount - The amount of crypto asset to sell, in its main unit (e.g., 0.5 for 0.5 ETH).
+   * @param {string} [refundAddress] - The wallet address to receive refunds in case of failure.
```

Comment:
What's the default value of this field? Include it in the description of the argument:
```suggestion
   * @param {string} [refundAddress] - The wallet address to receive refunds in case of failure (default: ?).
```